### PR TITLE
Fixes for various threading related crashes with rtsp netcams

### PR DESCRIPTION
**Pull request by jogu**
I've been running this set of commits for a number of days on my install now without any crashes, so I think it's finally good.

I recommend reviewing by looking at the diffs for each commit, as it'll make more sense then (ie. try to avoid looking at the overall diffs). It's quite a big set of changes, but I think any intermediate steps actually increase the number of crashes so I wouldn't want to merge anything other than the whole patch set. Each commit is quite small and makes sense on it's own though.

### CODE REVIEW

File: ffmpeg.c [hunk 1]
```diff
@@ -199,6 +239,13 @@ void ffmpeg_init(void){
     avcodec_register_all();
     avformat_network_init();
     av_log_set_callback((void *)ffmpeg_avcodec_log);
+
+    ret = av_lockmgr_register(ffmpeg_lockmgr_cb);
+    if (ret < 0)
+    {
+        MOTION_LOG(EMG, TYPE_ALL, SHOW_ERRNO, "%s: av_lockmgr_register failed (%d)", ret);
+        abort();
```
Comment (Mr-Dave) [inline ffmpeg.c]: Seems pretty drastic to just abend on them.  Can we get a return code and have a clean exit?

File: netcam_rtsp.c [hunk 2]
```diff
@@ -690,7 +702,7 @@ static int netcam_rtsp_resize(unsigned char *image , netcam_context_ptr netcam){
 
     retcd=my_image_copy_to_buffer(
          netcam->rtsp->swsframe_out
-        ,(uint8_t *)image
+        ,(uint8_t *)netcam->receiving->ptr
```
Comment (Mr-Dave) [inline netcam_rtsp.c]: This is causing a crash for me.  I think the image pointer has memory allocated to the size of the new image while the receiving pointer is sized to what the camera is sending.  To reproduce, set config parms to a bigger image than the camera.

File: netcam_rtsp.c [hunk 3]
```diff
@@ -935,14 +945,17 @@ int netcam_setup_rtsp(netcam_context_ptr netcam, struct url_t *url){
 */
 int netcam_next_rtsp(unsigned char *image , netcam_context_ptr netcam){
```
Comment (Mr-Dave) [inline netcam_rtsp.c]: With the elimination of the rtsp components in this function and the new information regarding what thread is calling this, should we move this function out of the netcam_rtsp.c module and over to the netcam.c?  It also now appears that we do not need the #IF on ffmpeg anymore.

Comment (jogu) [inline netcam_rtsp.c]: I really don't know :)

I think, given the function is specific to rtsp case (the one for the non-rtsp netcams works completely differently), keeping it in the rtsp specific file makes sense.

Removing the ifdefs makes sense. I've just pushed a commit that does that.

**Pull request merged**